### PR TITLE
ONEM-22435 Support LISA on reference image

### PR DESCRIPTION
**Pull request by sverkoye**
Found and fixed several issues while testing:
* build error: missing functional header include in Downloader.h
* add support for tar.gz in archiver
* fix problem extracting tars with hard links inside
* make sure downloaded file is closed (i.e. flushed) when
  starting extraction or else "truncated gzip input" errors
* during init: create lisa app dirs first, clean/prune afterwards
  (else crash when dirs do not exist in the first place)

### CODE REVIEW

File: LISA/Archive.cpp [hunk 1]
```diff
@@ -85,6 +86,12 @@ class Archive
             std::string destPath{destination + archive_entry_pathname(entry)};
             archive_entry_set_pathname(entry, destPath.c_str());
 
+            const char *orig_hardlink = archive_entry_hardlink(entry);
```
Comment (szymon-polke-red) [inline LISA/Archive.cpp:89]: pleasu use camelCase naming convention - origHardlink

File: LISA/Executor.cpp [hunk 2]
```diff
@@ -233,24 +233,25 @@ void Executor::doInstall(std::string type,
...
 
-    if (downloadSize > tmpFreeSpace) {
-        std::string message = std::string{} + "not enough space on " + tmpPath + " (available: "
+        if (downloadSize > tmpFreeSpace) {
+            std::string message = std::string{} + "not enough space on " + tmpPath + " (available: "
                 + std::to_string(tmpFreeSpace) +", required: " + std::to_string(downloadSize) + ")";
-        throw std::runtime_error(message);
+            throw std::runtime_error(message);
+        }
+        Filesystem::File tmpFile{tmpFilePath};
```
Comment (szymon-polke-red) [inline LISA/Executor.cpp:251]: tmpFile is only needed for download, so I suggest to modify Downloader.get to accept tmpFilePath and create tmpFile inside the function

File: LISA/Archive.cpp (hunk 1 above)
Comment (sverkoye) [reply LISA/Archive.cpp]: Done

File: LISA/Executor.cpp (hunk 2 above)
Comment (sverkoye) [reply LISA/Executor.cpp]: Done

**Review by szymon-polke-red — APPROVED** [review]

**Review by mariusz-sikora-red — APPROVED** [review]

**Pull request merged**